pkg/image/api/helper.go

```diff
@@ -191,6 +178,61 @@ func ParseDockerImageReference(spec string) (DockerImageReference, error) {
 	return ref, nil
 }
 
+// ParseDockerImageReference parses a Docker pull spec string into a
+// DockerImageReference.
+func ParseDockerImageReference(spec string) (DockerImageReference, error) {
+	// TODO replace with docker version once docker/docker PR11109 is merged upstream
+	stream, tag, id := parseRepositoryTag(spec)
+
+	ref, err := parseDockerImageReference(strings.Split(stream, "/"), spec)
+	if err != nil {
+		return ref, err
+	}
+	ref.Tag = tag
+	ref.ID = id
+
+	return ref, err
+}
+
+func ParseMultiSegmentsDockerImageReference(spec string) (DockerImageReference, error) {
+	stream, tag, id := parseRepositoryTag(spec)
+
+	repoParts := strings.Split(stream, "/")
+	if len(repoParts) < 4 {
```

Comment:
Right now we can have maximum 3 part (`registry/namespace/name`). Multi-segment path means  4 or more. So, old behavior will be in case of < 4.